tests/ha/hawk_gui.pm

```diff
@@ -92,60 +43,39 @@ sub run {
     }
 
     select_console 'root-console';
+    install_docker;
 
-    my $python = install_required_python_pkgs;
-    install_geckodriver $python;
-    # The line below can be changed for a call to lib/selenium.pm::enable_selenium_port()
-    # once Selenium::Remote::Driver it's available in the openqa.suse.de workers
-    assert_script_run('systemctl stop ' . opensusebasetest::firewall());
+    # TODO: Use another namespace in DockerHub
+    my $docker_image = "ricardobranco/hawk_test";
+    assert_script_run("docker pull $docker_image");
 
     select_console 'x11';
     x11_start_program('xterm');
     turn_off_gnome_screensaver;
 
-    # Download and prepare python selenium script
     my $pyscr = 'hawk_test';
-    my $path  = '/tmp';
-    download_selenium_script($pyscr, $path);
+    my $path  = 'test';
 
     # Run test
-    my $browser  = 'firefox';
-    my $version  = get_required_var('VERSION');
-    my $hostname = choose_node(1);
-    my $results  = "$path/$pyscr.results";
-    my $retcode  = "$path/$pyscr.ret";
-    my $logs     = "$path/$pyscr.log";
+    my $browser    = 'firefox';
+    my $version    = get_required_var('VERSION');
+    my $hostname   = choose_node(1);
+    my $results    = "$path/$pyscr.results";
+    my $retcode    = "$path/$pyscr.ret";
+    my $logs       = "$path/$pyscr.log";
+    my $virtual_ip = "10.0.2.222/24";
 
     add_to_known_hosts($hostname);
-    # Run the test
-    type_string "$python $path/$pyscr.py -b $browser -t $version -H $hostname -s $testapi::password -r $results > $logs 2>&1; echo $pyscr-\$? > $retcode; exit\n";
-    assert_screen "hawk-$browser", 60;
-
-    my $loop_count = 180;    # 15 minutes (180*5)
-    while (1) {
-        $loop_count--;
-        last if ($loop_count < 0);
-        if (check_screen('generic-desktop', 0, no_wait => 1)) {
-            # We may reach generic-desktop in two scenarios: (1) the python script
-            # finishes, or (2) it has finished an individual test and closed the
-            # browser but it's in the process of opening a new browser instance.
-            # The following check_screen tries to catch scenario 2, if it doesn't
-            # then we assume we're in scenario 1
-            next if (check_screen("hawk-$browser", 60));    # python script still running
-            last;
-        }
-        sleep 5;
-    }
-    if ($loop_count < 0) {
-        record_info("$browser failed", "Test with browser [$browser] could not be completed in 10 minutes", result => 'softfail');
-        send_key 'alt-f4';                                  # Force close of browser
-    }
+    assert_script_run "mkdir -m 1777 $path; touch $path/geckodriver.log $results";
+    my $docker_cmd = "docker run --rm --shm-size 128m -v \$PWD/$path:/$path $docker_image --xvfb -b $browser -t $version -H $hostname -s $testapi::password -r /$results --virtual-ip $virtual_ip";
+    assert_script_run "$docker_cmd | tee $logs; echo $pyscr-\$PIPESTATUS > $retcode", 1800;
```

Comment:
Did you try running the docker image from `user-console` instead of from an `xterm` window in `x11`? It may be possible if you use selenium's headless mode. You may even be able to drop the `is_desktop_installed` bit from the test.

Reply:
Will check that.